astronomer/providers/google/cloud/operators/bigquery.py

```diff
@@ -124,18 +124,23 @@ def execute(self, context: Context) -> None:  # noqa: D102
 
         self.job_id = job.job_id
         context["ti"].xcom_push(key="job_id", value=self.job_id)
-        self.defer(
-            timeout=self.execution_timeout,
-            trigger=BigQueryInsertJobTrigger(
-                conn_id=self.gcp_conn_id,
-                job_id=self.job_id,
-                project_id=self.project_id,
-                impersonation_chain=self.impersonation_chain,
-                poll_interval=self.poll_interval,
-                **kwargs,
-            ),
-            method_name="execute_complete",
-        )
+        if job.running():
+            self.defer(
+                timeout=self.execution_timeout,
+                trigger=BigQueryInsertJobTrigger(
+                    conn_id=self.gcp_conn_id,
+                    job_id=self.job_id,
+                    project_id=self.project_id,
+                    impersonation_chain=self.impersonation_chain,
+                    poll_interval=self.poll_interval,
+                    **kwargs,
+                ),
+                method_name="execute_complete",
+            )
+        else:
+            job.result()
+            self._handle_job_error(job)
+            return self.job_id
```

Comment:
if we are processing the job result here and returning job_id, we should also do the same in execute_complete to having consistent results in the XCOM for the task.

Reply:
Yep, that makes sense. Just update it. Thanks!